tests/integration/ambient/cacert_rotation_test.go

```diff
@@ -0,0 +1,176 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ambient
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+
+	"istio.io/istio/istioctl/pkg/util/configdump"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	kubetest "istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/util/assert"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/security/pkg/pki/util"
+	"istio.io/istio/tests/integration/security/util/cert"
+)
+
+func TestIntermediateCertificateRefresh(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.peer.cacert-rotation").
+		Run(func(t framework.TestContext) {
+			istioCfg := istio.DefaultConfigOrFail(t, t)
+			istioCtl := istioctl.NewOrFail(t, t, istioctl.Config{})
+			namespace.ClaimOrFail(t, t, istioCfg.SystemNamespace)
+			newX509 := getX509FromFile(t, "ca-cert-alt.pem")
+
+			sa := apps.Captured[0].ServiceAccountName()
+
+			// we do not know which ztunnel instance is located on the node as the workload, so we need to check all of them initially
+			ztunnelPods, err := kubetest.NewPodFetch(t.AllClusters()[0], istioCfg.SystemNamespace, "app=ztunnel")()
+			assert.NoError(t, err)
+
+			originalWorkloadSecret, ztunnelPod, err := getWorkloadSecret(t, ztunnelPods, sa, istioCtl)
+			if err != nil {
+				t.Errorf("failed to get initial workload cert: %v", err)
+			}
+
+			// Update CA with new intermediate cert
+			if err := cert.CreateCustomCASecret(t,
+				"ca-cert-alt.pem", "ca-key-alt.pem",
```

Comment:
@GregHanson This is totally new root cert

Reply:
correct, but `root-cert-combined.pem` contains both `root-cert.pem` and `root-cert-allt.pem`?  What I really needed was a second set of `-alt.pem` files, generated using the same `root-cert.pem` file.  The `CreatCASecret` setup func isn't configurable though.  Also the `root-cert.pem` in `samples/certs` is generated by `security/samples/plugin_ca_certs/gen_certs.sh` and I can't get them to plugin to `tools/certs/Makefile.selfsigned.mk` used to generate alt intermediate certificates.

To get around this, I created a new `CreateCASecretAlt` setup func so I can specify alt certs to use at deploy time